vividus-bdd-engine/src/main/java/org/vividus/bdd/transformer/ResolvingSelfReferencesEagerlyTransformer.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.transformer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.inject.Named;
+
+import org.jbehave.core.model.ExamplesTable.ExamplesTableProperties;
+import org.jbehave.core.model.TableParsers;
+import org.jbehave.core.steps.ParameterControls;
+import org.vividus.bdd.util.ExamplesTableProcessor;
+
+@Named("RESOLVING_SELF_REFERENCES_EAGERLY")
+public class ResolvingSelfReferencesEagerlyTransformer implements ExtendedTableTransformer
+{
+    private final ParameterControls parameterControls;
+
+    public ResolvingSelfReferencesEagerlyTransformer(ParameterControls parameterControls)
+    {
+        this.parameterControls = parameterControls;
+    }
+
+    @Override
+    public String transform(String tableAsString, TableParsers tableParsers, ExamplesTableProperties properties)
+    {
+        List<String> tableAsRows = ExamplesTableProcessor.parseRows(tableAsString);
+        List<String> header = tableParsers.parseRow(tableAsRows.get(0), true, properties);
+        List<List<String>> inputRows = ExamplesTableProcessor.parseDataRows(tableAsRows, tableParsers, properties);
+        SelfReferencesResolver resolver = new SelfReferencesResolver();
+        List<List<String>> resolvedRows = inputRows.stream().map(dataRow -> resolver.resolveRow(dataRow, header))
+                .collect(Collectors.toList());
+        return ExamplesTableProcessor.buildExamplesTable(header, resolvedRows, properties, true);
+    }
+
+    private final class SelfReferencesResolver
+    {
+        private Map<String, String> valuesToResolve;
+        private Map<String, String> resolvedValues;
+        private final Pattern placeholderPattern;
+
+        private SelfReferencesResolver()
+        {
+            placeholderPattern = Pattern.compile(addDelimiters("(.*?)"));
```

Comment:
this should be converted to a final field of transformer to avoid pattern re-compilation for each table row